code/game/objects/items/weapons/material/misc.dm

```diff
@@ -93,3 +93,112 @@
+	slot_flags = SLOT_BELT
+	w_class = ITEMSIZE_NORMAL
+	origin_tech = list(TECH_COMBAT = 2)
+	attack_verb = list("flogged", "whipped", "lashed", "disciplined")
+	force_divisor = 0.15
+	thrown_force_divisor = 0.25
+	reach = 2
+
+	item_icons = list(
+			slot_l_hand_str = 'icons/mob/items/lefthand_melee.dmi',
+			slot_r_hand_str = 'icons/mob/items/righthand_melee.dmi',
+			)
+
+/obj/item/weapon/material/whip/afterattack(atom/A as mob|obj|turf|area, mob/user as mob, proximity)
+	..()
+
+	if(!proximity)
+		return
+
+	if(istype(A, /atom/movable))
+		var/atom/movable/AM = A
+		if(AM.anchored)
+			to_chat(user, "<span class='notice'>\The [AM] won't budge.</span>")
+			return
+
+		else
+			if(!istype(AM, /obj/item))
+				user.visible_message("<span class='warning'>\The [AM] is pulled along by \the [src]!</span>")
+				AM.Move(get_step(AM, get_dir(AM, src)))
+				return
+
+			else
+				user.visible_message("<span class='warning'>\The [AM] is snatched by \the [src]!</span>")
+				AM.throw_at(user, reach, 0.1, user)
+
+/obj/item/weapon/material/whip/apply_hit_effect(mob/living/target, mob/living/user, var/hit_zone)
+	if(user.a_intent)
+		switch(user.a_intent)
+			if(I_HURT)
+				if(prob(10) && istype(target, /mob/living/carbon/human))
+					to_chat(target, "<span class='warning'>\The [src] rips at your hands!</span>")
+					ranged_disarm(target)
+			if(I_DISARM)
+				if(prob(min(90, force * 3)) && istype(target, /mob/living/carbon/human))
+					ranged_disarm(target)
+				else
+					target.visible_message("<span class='danger'>\The [src] sends \the [target] stumbling away.</span>")
+					target.Move(get_turf(get_step(target,get_dir(user,target))))
+			if(I_GRAB)
+				var/turf/STurf = get_turf(target)
+				spawn(2)
+					playsound(STurf, 'sound/effects/snap.ogg', 60, 1)
+				target.visible_message("<span class='critical'>\The [src] yanks \the [target] towards \the [user]!</span>")
+				target.throw_at(get_turf(get_step(user,get_dir(user,target))), 2, 1, src)
+
+	..()
+
+/obj/item/weapon/material/whip/proc/ranged_disarm(var/mob/living/carbon/human/H, var/mob/living/user)
+	if(istype(H))
+		var/list/holding = list(H.get_active_hand() = 40, H.get_inactive_hand() = 20)
+
+		for(var/obj/item/weapon/gun/W in holding)
+			if(W && prob(holding[W]))
+				var/list/turfs = list()
+				for(var/turf/T in view())
+					turfs += T
+				if(turfs.len)
+					var/turf/target = pick(turfs)
+					visible_message("<span class='danger'>[H]'s [W] goes off due to \the [src]!</span>")
+					return W.afterattack(target,H)
+
+		if(!(H.species.flags & NO_SLIP) && prob(10))
+			var/armor_check = H.run_armor_check(user.zone_sel, "melee")
+			H.apply_effect(3, WEAKEN, armor_check)
+			playsound(src, 'sound/weapons/thudswoosh.ogg', 50, 1, -1)
+			if(armor_check < 60)
+				visible_message("<span class='danger'>\The [src] has tripped [H]!</span>")
+			else
+				visible_message("<span class='warning'>\The [src] attempted to trip [H]!</span>")
+			return
+
+		else
+			if(H.break_all_grabs(user))

```

Comment:
So if you fail to "attempt" to trip them (at a 10% chance), you intead make everyone stop grabbing them? Even if you're just trying to beat their face in?

Reply:
It cuts off _their_ grabs on others. People will still be holding onto them if they are struck.